### PR TITLE
Support using `null` without explicit cast in UNION

### PR DESCRIPTION
**Pull request by mfussenegger**
Adds support for cases like `select null union all select x from tbl` (or the other way around)

We could also support implicit casts as requested in  https://github.com/crate/crate/issues/12511 but I'm not sure if it is a good idea.

Would only have to adjust the logic here to check for cast compatibility instead of checking if they're compatible:

https://github.com/crate/crate/blob/76c619b12a5157592d2e6db67c3bafc089568287/server/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java#L226-L244

The remainder of the logic should then already take care of the rest.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java [hunk 1]
```diff
@@ -1651,4 +1651,19 @@ public void test_insert_into_table_with_primary_key_with_default_clause_sets_rig
             "1\n"
         ));
     }
+
+    @Test
+    public void test_insert_from_union_with_null_literal() {
```
Comment (seut) [inline server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java:1656]: does this test cover any special branch not covered by the analyzer test? I wonder if it is really needed.

**Review by seut — APPROVED** [review]
lgtm otherwise

File: server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java:1656]: It triggered an assertion in the union operator and covers the changes in it, but I can turn it into a planner test

Comment (seut) [reply server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java:1656]: ah right. planner test would be great if no big thing.

**Pull request merged**